apps/api/src/users/users.controller.ts

```diff
@@ -36,4 +52,74 @@ export class UsersController {
   ): Promise<AccessTokenResponse> {
     return this.usersService.updateUserPassword(user, updateUserPasswordDto);
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Header('Content-Type', 'image/jpeg')
+  @Get(':id/profile-picture')
+  async getUserProfilePicture(
+    @Param('id') id: string,
+  ): Promise<StreamableFile> {
+    const profile = await this.usersService.getProfile(id);
+    if (profile?.picture == null) {
+      throw new NotFoundException(
+        `User: ${id} does not have a profile picture`,
+      );
```

Comment:
It would be easier for the front-end to send the default image as is, instead of throwing an error.